### DISCUSSION TITLE
Questions about img loading via emails.

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#68069] [Help]
### Summary

I am sending this piece of code  in the html section of nodemailer.  The email is sent properly, but the src doesn't seem to be loading or communicating back to my computer when testing from my computer.

The image code is:
`<div><img src="${img}" alt="quote" style="width: 100%; height: auto;"/></div`

and the src for the image would be: `http://localhost:3000/my/file/1`

^^^  when I type the src address in my computer, the code executes automatically and sends me another (second) email to confirm it is working...  but when I check in my gmail, and I open the original email where the image was sent, it seems like the src code is doing nothing as I don't receive the second email confirmation and apparently the src code isn't loading?

What am I missing? Could it be that Gmail is blocking the src code?   When I check the element code from my Gmail it looks like this:

`<img src="https://ci3.googleusercontent.com/meips/ADKq_NbJ7q6h17hYujM8RomUzMzHfxGGB4mJu2MtQ1uBc5NF_vIU-w2iOkr6N8JjqnCMVI4LjcEa=s0-d-e1-ft#http://localhost:3000/my/file/2" alt="quote" style="width:100%;height:auto" class="CToWUd" data-bit="iit" jslog="138226; u014N:xr6bB; 53:WzAsMl0.">`

**Comment by knarrff** [comment]
Hi @Elindo586 

Most email clients are by default configured to not open images at all, or only after an explicit question if this is wanted. Even if images are rendered, this typically is only allowed for images that came with the email (embedded or as attachments), i.e., explicitly not using remote URLs.
This is by design: it otherwise would be too easy to breach privacy of the user by not only confirming that the email was read, but also when, how often, using which client, and from which IP address. This is also why every user should use strict settings here and why something like you try will not work, at least for most users.

Having said that, the issue at hand here is with gmail, since that is your email client in this case. Google routes all images through their usercontentproxy, thus the URL changes. Now, using "localhost" in emails is not a great idea to begin with (since that resolves to a different machine to each client), and here this also shows: google will try to fetch the image from "localhost", which of course for google, isn't your machine (and likely google will not even try, seeing that this is 'localhost' and cannot possibly be correct). In order to make this work with gmail (and also for a general user base), the URL has to point to a publicly accessible server.

**Reply by Elindo586** [reply]
Thank you much for your reply.

The localhost is only for testing purposes. 

I was wondering how companies like Constant Contact track their email campaigns, and I was trying to do something similar.

I was also reading that many email provider are now blocking pixel tracking, so I might not be able to track via pixels.

On a similar situation from emails I receive, I see images are blocked, when I allow the downloading I can see a tracking web address, and when I click on it, it re-direct me to a different page.

^^  To do this above it might just be an `<a>` tag with an `<img>` tag and `href` for the tracking, and a re-direct to a page in the front end. 


Still, how can anybody claim to track just an opening of an email when many email providers are now blocking it?  hmm.. I don't think they can accurately do it.
